internal/server/auth/middleware.go

```diff
@@ -37,27 +44,57 @@ func GetAuthenticationFrom(ctx context.Context) *authrpc.Authentication {
 	return auth.(*authrpc.Authentication)
 }
 
+// InterceptorOptions configure the UnaryInterceptor
+type InterceptorOptions struct {
+	skippedServers []any
+}
+
+func (o InterceptorOptions) skipped(server any) bool {
+	for _, s := range o.skippedServers {
+		if s == server {
+			return true
+		}
+	}
+
+	return false
+}
+
+// WithServerSkipsAuthentication can be used to configure an auth unary interceptor
+// which skips authentication when the provided server instance matches the intercepted
+// calls parent server instance.
+// This allows the caller to registers servers which explicitly skip authentication (e.g. OIDC).
+func WithServerSkipsAuthentication(server any) containers.Option[InterceptorOptions] {
```

Comment:
is there anyway to identify a server by name instead of doing pointer comparison (when checking for if it's in the `o.skippedServers[]`)?

Reply:
I think there might be. We could do a comparison on the string of the full method: https://pkg.go.dev/google.golang.org/grpc#UnaryServerInfo

However, I felt this was relatively more concrete. Since it is direct pointer comparison with the instance we want to skip auth on. We compare with the method name and then someone renames a package or type, and it moves under us.
While this does use `any` you still have to pass it something and if that something gets renamed it won't compile until you correct it.